docs/fpc-key-dist.puml

```diff
@@ -1,97 +1,319 @@
+    can proceed in the case the running enclave is
+    unavailable - for any possible or malicious cause.
+    Key confirmation can ensure that backup enclave are
+    ready to take over and/or contribute.
+  end note
+  group key confirmation endorsement
+    Admin1  -> Admin1       :   m <- <ECC1_S_PK, SIG_ECC1, CC_E_PK>
+    Admin1  -> ERCC1        ++: ConfirmCCKeys(m)
+    ERCC1   -> ERCC1        :   check that creator is Admin
+    ERCC1   -> ERCC1        :   check that admin.org and enclave.org match
+    ERCC1   -> ERCC1        :   check ECC1_S_PK is registered
+    ERCC1   -> ERCC1        :   verify SIG_ECC1 over <CC_E_PK> using ECC1_S_PK
+    ERCC1   -> TLCC_Enclave1++: can_endorse(ECC1_S_PK)
+        note left
+            A secure channel with TLCC
+            is not required here.
+        end note
+    return true
+    note left
+        if false, abort
+    end note
+    ERCC1   -> ERCC1    :   put_state(key=<CC_name, CC_ver, ECC1_S_PK, "keydist_confirmed">, value=m)
+        note left
+            This is the proof (stored on the ledger)
+            that the enclave has the CC keys.
+            In a multi-org setting, the proof informs
+            of enclave availability.
+        end note
+
+    return /'Admin1 -> ERCC1'/
+
+  end group
+
+  ref over Admin1, ERCC2
+    key confirmation endorsement
+  end ref
+
+  Admin1 -> Orderer: submit key confirmation transaction
 
-  Admin -> ECC_Manager1 ++: generate key (CC_name, CC_ver, CH_id)
+end group
 
-  ECC_Manager1 -> ECC1 ++: generate key
 
-  ECC1 -> Enclave1 ++: generate key
+group key distribution
+  ref over Admin1, TLCC_Enclave1, Admin2, TLCC_Enclave2
+    if necessary, upgrade chaincode policy to enable Enclave2 endorsements
+  end ref 
+  
+  ref over Admin2, ECC_Manager2, ECC2, Enclave2, ERCC2, TLCC_Enclave2
+    enclave creation, attestation generation, register attestation (see fpc-registration-new.puml)
+  end ref
 
-  Enclave1 -> Enclave1 : S_State, CC_{PK,SK} <- keyGen
-  note right
-    - S_State denotes state secret key.
-    - CC_{PK,SK} denotes chain code pub/priv key pair
+  note over Admin1, Admin2
+    At this point, a chaincode enclave is up and running, and another is waiting for the chaincode keys.
+    We need a means to trigger the key distribution. Now, an admin does it.
+    Can Fabric automate this?
   end note
 
-  Enclave1 -> Enclave1 : m <- encrypt {{S_State, CC_SK}} with ECC_E_PK
-  note right
-    enclave encryptes state secret key and chaincode privkey.
-    - m denotes encrypted message
+  Admin1        -> ERCC1            ++: query registered enclave's key
+    note right
+        ERCC will likely return the list of enclaves
+        with their cryptographic material. Admin1
+        then selects the non-provisioned one(s).
+    end note
+  return ECC2_S_PK /'Admin1 -> ERCC1'/
+
+  Admin1        -> Admin1           :   m <- <CC_name, CC_ver, CH_id, org_id, ECC2_S_PK>
+  Admin1        -> ECC_Manager1     ++: DistributeKey(m)
```

Comment:
same comment here, I suggest that instead of `DistributeKey(m)` we use `ShareCCKeys(target, cc_ctx)` where `target` is `ECC2_S_PK` and `cc_ctx` is the cc_name, etc ... 

This also matches better `GenerateCCKeys()`

Reply:
fixed consistency and separated the paramenters